### PR TITLE
feat(Highlighter): add highlightStyles to find method

### PR DESCRIPTION
**Pull request by mukiienko**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. Use imperative, present tense in your commit description ("change", not "changed" or "changes") without uppercases or period (.) at the end.

# Checklist
- [x] The implementation has been manually tested and complies with Textkernel [browser support guidelines](https://textkernel.com/browser-support/)
- [x] The implementation complies with [accessibility](CONTIRBUTING.md#accessibility) standards.
- [x] The component has a [displayName](CONTRIBUTING.md#display-names) defined.
- [x] The component comes with a detailed [PropTypes](CONTRIBUTING.md#component-props) (and defaultProps) definition.
- [x] Component PropTypes are sufficiently described / documented.
- [x] There is [a story](CONTRIBUTING.md#component-showcases) in Storybook.

### CODE REVIEW

File: src/packages/Highlighter/Highlighter.ts [hunk 1]
```diff
@@ -196,14 +196,18 @@ export class Highlighter {
...
+        highlightStyles?: CSSStyleDeclaration
+    ) {
         const matchedTerms: HighlighterResultItem[] = [];
         const notMatchedTerms: HighlighterResultItem[] = [];
         const sortedTerms = HighlighterCore.sortTerms(terms);
 
         sortedTerms.forEach((term, index) => {
             let amount = 0;
-            const styles = Highlighter.generateHighlightStyles(index);
+            const styles = highlightStyles || Highlighter.generateHighlightStyles(index);
```
Comment (agorodetskaya) [inline src/packages/Highlighter/Highlighter.ts:210]: what do you think about overriding styles, not replacing:
```
const styles = {
  ...Highlighter.generateHighlightStyles(index),
  ...highlightStyles,
};
```

**Comment by agorodetskaya** [comment]
@mukiienko as I understand Reach needs to have single highlight color for all terms. Maybe it is better to implements such flag or something directly related to colors and do not allow to fully override styles?

File: src/packages/Highlighter/Highlighter.ts (hunk 1 above)
Comment (mukiienko) [reply src/packages/Highlighter/Highlighter.ts:210]: `generateHighlightStyles` returns the bunch of styles that actually redundant for the current purpose - having the only background color. 
```
    border-radius: 1px;
    display: inline-block;
    text-indent: initial;
    background-color: rgb(214, 244, 129);
```

**Comment by mukiienko** [comment]
I would like to keep the ability to control all styles. It is also consistent with `StringHightligher` that accepts any styles. So, if we want to make highlighted terms bold it will be easy to tweak on Reach side without amendment OneUI

File: src/packages/Highlighter/Highlighter.ts (hunk 1 above)
Comment (agorodetskaya) [reply src/packages/Highlighter/Highlighter.ts:210]: ` border-radius: 1px;` is'n it useful one?
So if we just need to change color - we also have to send other styles to have consistent style everywhere in TK

**Comment by agorodetskaya** [comment]
I just had doubts about having different highlighters across TK. For example some teams add border-radius and other not.
But oki, I don't mind to have styles here, moreover if we have it for StringHightligher.

File: src/packages/Highlighter/Highlighter.ts (hunk 1 above)
Comment (mukiienko) [reply src/packages/Highlighter/Highlighter.ts:210]: The `border-radius` looks fancy for the overview profile where we have bright colors in the background and use the border for selected terms and elements should be `inline-block`. For yellow highlighting (string), it doesn't make sense.

**Comment by mukiienko** [comment]
@agorodetskaya I see your point, but if we can still change the color it will be breaking consistency between teams anyway. Initially, when we started to develop OneUI library the idea was to have a public library that is not tightly coupled with our products, and other users can reuse our components.

**Review by agorodetskaya — APPROVED** [review]

**Review by carlobernardini — APPROVED** [review]

**Pull request merged**